utils/text_extractor.py

```diff
@@ -47,11 +53,22 @@ def get_all_text_from_url(url, retries=5):
         allText = h.handle(html)
 
         return allText, title
+    except timeout as exc:
+        logging.error(
+            "an exception occurred while trying to access url {} trying again \n more details: {}".format(url, exc))
+        time.sleep(1)
+        retries = retries - 1
+        if retries == 0:
+            increase_timeout_count(url)
+            raise timeout("The request to the URL has timed out.")
+        return get_all_text_from_url(url, retries)
```

Comment:
ok, I see it here.
not critical but!
usually, there is a "business level" exception module (located inside the implemented module)
so you will have something like TextExtraction(Exception) 

why? Because it helps to decuple our logic from the 3rd party we use

Reply:
you mean creating an exception class for TextExtraction that extends the default exception class so we can raise exceptions that are more explicit? I have only done this in java but I can try:)